### PR TITLE
Neox 2.x

### PR DESCRIPTION
**Pull request by KickSeason**
This is a version to support cross chain on neo-2.x. 
* root in consensus
* p2p support of root
* blockchain with root
* new interop service  `Neo.CrossChain.Keccak256` `Neo.CrossChain.Ecrecover`
* rpc for root related

### CODE REVIEW

**Comment by shargon** [comment]
Could you create first one for format the code, it will be easier for review the changes

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -219,12 +224,16 @@ private void OnCommitReceived(ConsensusPayload payload, Commit commit)
...
+                byte[] hashData1 = context.MakeHeader()?.GetHashData();
+                byte[] hashData2 = context.MakeStateRoot()?.GetHashData();
+                if (hashData1 == null || hashData2 == null)
                 {
                     existingCommitPayload = payload;
                 }
-                else if (Crypto.Default.VerifySignature(hashData, commit.Signature,
+                else if (Crypto.Default.VerifySignature(hashData1,
+                    commit.Signature,
+                    context.Validators[payload.ValidatorIndex].EncodePoint(false)) && Crypto.Default.VerifySignature(hashData2,
```
Comment (vncoelho) [inline neo/Consensus/ConsensusService.cs:235]: looks good.

**Comment by shargon** [comment]
Could you move the format to other PR in order to reduce the changes?

**Comment by KickSeason** [comment]
@shargon @erikzhang I  close this and  divide into multi pull requests.

**Pull request closed without merging**